Tasks/AzureFileCopyV2/AzureFileCopyRemoteJob.ps1

```diff
@@ -0,0 +1,108 @@
+$AzureFileCopyRemoteJob = {
+    param(
+        [string]$containerURL,
+        [string]$targetPath,
+        [string]$containerSasToken,
+        [string]$additionalArguments,
+        [string]$azCopyToolFileNamesString,
+        [string]$azCopyToolFileContentsString,
+        [switch]$CleanTargetBeforeCopy,
+        [switch]$EnableDetailedLogging
+    )
+
+    function Write-DetailLogs
+    {
+        [CmdletBinding()]
+        param(
+            [string]$message
+        )
+
+        if($EnableDetailedLogging)
+        {
+            Write-Verbose $message
+        }
+    }
+
+    function Get-AzCopyVerboseLogs
+    {
+        [CmdletBinding()]
+        param(
+            [bool]$isLogsPresent,
+            [string]$logFilePath
+        )
+
+        if($isLogsPresent)
+        {
+            Write-DetailLogs (Get-Content -Path $logFilePath | Out-String)
+        }
+    }
+
+    function Remove-AzCopyFolder
+    {
+        [CmdletBinding()]
+        param([string]$azCopyLocation)
+
+        $tempParentFolder = (Get-Item $azCopyLocation).Parent.FullName
+
+        Get-ChildItem -Path $azCopyLocation -Recurse -Force | Remove-Item -Force -Recurse
+        Remove-Item $azCopyLocation -Force
+        Remove-Item $tempParentFolder -Force
+    }
+
+    try
+    {
+        $azCopyToolFileNames = $azCopyToolFileNamesString.Split(";")
+        $azCopyToolFileContents = $azCopyToolFileContentsString.Split(";")
+
+        $randomFolderName = "AFC_" + [guid]::NewGuid()
+        $randomFolderPath = Join-Path -Path $env:windir -ChildPath "DtlDownloads\$randomFolderName"
```

Comment:
add verbose logging as creating temp folder, creating azcopy tools, will be helpfull in debuggin

Reply:
done